src/derivers/bip32.ts

```diff
@@ -111,29 +102,55 @@ export async function deriveChildKey(
     );
   }
 
-  const secretExtension = await deriveSecretExtension({
-    parentPrivateKey: parentKey as Buffer,
-    childIndex,
-    isHardened,
-    curve,
-  });
-
-  const { privateKey, extraEntropy } = generateKey({
-    parentPrivateKey: parentKey as Buffer,
-    parentExtraEntropy: chainCode,
-    secretExtension,
-    curve,
-  });
-
-  return [
-    Buffer.from(privateKey),
-    await curve.getPublicKey(privateKey),
-    Buffer.from(extraEntropy),
-  ];
+  if (privateKey) {
+    if (privateKey.length !== BUFFER_KEY_LENGTH) {
+      throw new Error('Invalid parent key: Must be 32 bytes long.');
+    }
+
+    const secretExtension = await deriveSecretExtension({
+      privateKey,
+      childIndex,
+      isHardened,
+      curve,
+    });
+
+    return await generateKey({
+      privateKey,
+      chainCode,
+      secretExtension,
+      curve,
+    });
+  }
+
+  if (publicKey) {
+    if (publicKey.length !== curve.publicKeyLength) {
+      throw new Error(
+        `Invalid parent public key: Must be ${curve.publicKeyLength} bytes long.`,
+      );
+    }
+
+    const compressedPublicKey = curve.compressPublicKey(publicKey);
```

Comment:
Do we always want to compress the public key?

Reply:
Yeah, the child public key is derived from the compressed parent public key.
- https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#conventions
- https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki#public-parent-key--public-child-key